app/dturn/dturn.js

```diff
@@ -176,437 +181,444 @@ function sortLargestFirst(a, b) {
-    }
-    if (!Discussions[discussionId]?.ShownStatements?.length) {
-        return undefined
-    }
-    if (Discussions[discussionId].ShownStatements?.[0].length < Discussions[discussionId].group_size * 2 - 1)
-        return undefined
-    const dis = Discussions[discussionId]
-    const statementIds = []
-    let authoredId // id of statment the user authored -- if the shownGroup is incomplete
-    if (dis.Uitems?.[userId]?.[round]) {
-        const sIds = Object.keys(dis.Uitems[userId][round].shownStatementIds)
-        if (round === 0 && sIds.length === 1 && dis.Uitems[userId][round].shownStatementIds[sIds[0]].author) {
-            statementIds.push(sIds[0])
-            authoredId = sIds[0]
-        } else if (sIds.length >= dis.group_size) {
-            console.error('user has been here before', discussionId, userId, round)
-            return sIds
-        } else
-            throw new Error(
-                `getStatments unexpected number of statments ${JSON.stringify(dis.Uitems?.[userId]?.[round], null, 2)}`
-            )
-    }
-    if (dis.ShownGroups[round]?.at(-1)?.shownCount < Math.pow(dis.group_size, round + 1)) {
-        if (authoredId && dis.ShownGroups[round].at(-1).statementIds.some(id => id === authoredId)) return // the user's statement is in the ShownGroup
-        for (const sId of dis.ShownGroups[round].at(-1).statementIds) statementIds.push(sId)
-        dis.ShownGroups[round].at(-1).shownCount++
-    } else if (round === 0) {
-        // find all the statments that need to be seen, and randomly pick GROUP_SIZE-1 -- because the user will add one of their own
-        const needToBeSeen = dis.ShownStatements[round].filter(
-            sItem => sItem.shownCount < Math.pow(dis.group_size, round + 1)
-        ) //??? Should this GROUP_SIZE increase in situations where there are lots of similar ideas that get grouped - but not in round 0
-        const shownGroup = { statementIds: [], shownCount: 0 }
-        if (needToBeSeen.length < dis.group_size - 1) return // don't create irregular size groups
-        else if (needToBeSeen.length == dis.group_size - 1) {
-            // exactly enough
-            for (const sItem of needToBeSeen) {
-                statementIds.push(sItem.statementId)
-                shownGroup.statementIds.push(sItem.statementId)
-            }
-        } else {
-            getRandomUniqueList(needToBeSeen.length, dis.group_size - 1).forEach(index => {
-                statementIds.push(needToBeSeen[index].statementId)
-                shownGroup.statementIds.push(needToBeSeen[index].statementId)
-            })
-        }
-        if (!dis.ShownGroups[round]) dis.ShownGroups[round] = [] // don't create the blank until theres somthing to put there so showdeepdiff works after reconstituting
-        dis.ShownGroups[round].push(shownGroup)
-        if (authoredId && shownGroup.statementIds.some(id => id === authoredId)) return // the user's statement is in the ShownGroup
```

Comment:
This like was the problem.  If get statements got the statement that the user inserted, it would just return undefined.  Since we were doing this with a random number generator it worked sometimes and not other times. 

I fixed it by filtering out the users statement above before picking at random.

I also merged in some code from master where allows us to set an environment variable and then not user random numbers.